### PR TITLE
New Demonstration Alpha Models

### PR DESCRIPTION
**Pull request by simonsonjack**
#### Description
I've updated the order ticket demo algorithms to better reflect best/standard industry practices. The new alpha models cover a variety of topics and details can be found in the comments at the top of each file. Deleted the ShareClassMeanReversion alpha models currently existing since these will be replaced with a version that is built as a Framework Algo. rather than a Framework Bridge Algo.

#### Motivation and Context
Order ticket updates come from some community user notifications and it will update our examples to best represent how Limit Orders and Stop Limit Orders are generally used.

#### How Has This Been Tested?
All tested locally and in the cloud. Verified results via comparison of logs and consultation with other developers.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+
+class ContingentClaimsAnalysisAlphaModel:
+    
+    def __init__(self, *args, **kwargs):
+        self.symbolDataBySymbol = {}
+        self.month = None
+        self.default_threshold = kwargs['default_threshold'] if 'default_threshold' in kwargs else 0.25
+        self.expiry = None
+        self.start_time = kwargs['start_time'] if 'start_time' in kwargs else None
```
Comment (AlexCatarino) [inline Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py]: If `self.start_time` is `None`, `tau` calculation in line 160 will give us:
```
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'NoneType'
```
So I think it should default to today:
```python
self.start_time = kwargs['start_time'] if 'start_time' in kwargs else \
    datetime.combine(date.today(), datetime.min.time())
```

File: Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+            
+        ## Build a list to hold our insights
+        insights = []
+        
+        for symbol, symbolData in self.symbolDataBySymbol.items():
+            pod = self.CCADefaultProbability(algorithm, symbolData)
+            algorithm.Log('POB for ' + str(symbol) + ': ' + str(pod))
+            ## If Prob. of Default is greater than our set threshold, then emit an insight indicating that this asset is trending downward
+            if (pod >= self.default_threshold) and (pod != 1.0):
+                insights.append(Insight(symbol, timedelta(days = 30), InsightType.Price, InsightDirection.Down, pod, None))
```
Comment (AlexCatarino) [inline Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py]: Just a comment, not a required change:
We can use the static helper method `Insight.Price` and the insight period could be configurable:
```python
insights.append(Insight.Price(symbol, self.period, InsightDirection.Down, pod, None))
```
where `self.period` is defined in `__init__` as `self.period = timedelta(days = 30)`.

File: Algorithm.Python/Alphas/GasAndCrudeOilEnergyCorrelationAlpha.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+    
+    def __init__(self, *args, **kwargs):
+        
+        self.difference_trigger = kwargs['difference_trigger'] if 'difference_trigger' in kwargs else 2.5
+        self.lookback = kwargs['lookback'] if 'lookback' in kwargs else 5 ## In hours
+        self.history_days = kwargs['history_days'] if 'history_days' in kwargs else 90 ## In days
+        self.resolution = kwargs['resolution'] if 'resolution' in kwargs else Resolution.Hour
+        self.prediction_interval = Time.Multiply(Extensions.ToTimeSpan(self.resolution), 5) ## Arbitrary
+        self.symbolDataBySymbol = {}
+        self.pairs_tickers = kwargs['pairs_tickers'] if 'pairs_tickers' in kwargs else None
```
Comment (AlexCatarino) [inline Algorithm.Python/Alphas/GasAndCrudeOilEnergyCorrelationAlpha.py]: `self.pairs_tickers` cannot default to `None`.

File: Algorithm.Python/Alphas/MultipleShareClassMeanReversionAlpha.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+class ShareClassMeanReversionAlphaModel(AlphaModel):
+
+    def __init__(self, *args, **kwargs):
+        ''' Initialize variables and dictionaries to help with SMA and Symbol Date
+        object creation/maintenance'''
+        self.symbolDataBySymbol = {}
+        self.sma_period = 15
+        self.period = timedelta(minutes=5)
+        self.number_pairs = kwargs['number_pairs'] if 'number_pairs' in kwargs else None
```
Comment (AlexCatarino) [inline Algorithm.Python/Alphas/MultipleShareClassMeanReversionAlpha.py]: `self.ticker_list`, `self.ticker_pairs` and `self.number_pairs` cannot default to `None`. For the last attribute, we can use `len(self.ticker_pairs)` instead:
```python
if 'ticker_pairs' not in kwargs:
    raise Exception('ShareClassMeanReversionAlphaModel: Missing argument: "ticker_pairs"')
self.ticker_pairs = kwargs['ticker_pairs']
self.number_pairs = kwargs['number_pairs'] if 'number_pairs' in kwargs else len(self.ticker_pairs)
```

File: Algorithm.Python/Alphas/ArtificiallyDepressedPricesAlpha.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+           data.Delistings.ContainsKey(self.Symbol) or \
+           data.SymbolChangedEvents.ContainsKey(self.Symbol):
+            return False
+        
+        ## Update close, volume, and SMAs
+        bar = data[self.Symbol]
+        self.Close = bar.Close
+        self.Volume = bar.Volume
+        self.CloseSMA.Update(bar.EndTime, bar.Close)
+        self.VolumeSMA.Update(bar.EndTime, bar.Volume)
```
[the remark this answers is not in the document]
Comment (simonsonjack) [reply Algorithm.Python/Alphas/ArtificiallyDepressedPricesAlpha.py]: This hasn't been the case thus far, but I'll make the changes

**Pull request closed without merging**